exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java

```diff
@@ -56,6 +58,31 @@ protected void readField(long recordsToReadInThisPass) {
     }
```

Comment:
What about NullableDictionaryUint types?
Also, It is probably a good idea to create a single parquet file with columns of all combinations of these types, 
Nullable, dictionary encoded, signed and unsigned int2, int4, int8 types and a single unit test to read that file.